alacritty/src/display/mod.rs

```diff
@@ -1371,23 +1377,38 @@ impl Display {
     }
 
     /// Requst a new frame for a window on Wayland.
-    #[inline]
-    #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
-    fn request_frame(&self, window: &Window) {
-        let surface = match window.wayland_surface() {
-            Some(surface) => surface,
-            None => return,
-        };
+    fn request_frame(&mut self, scheduler: &mut Scheduler) {
+        // Mark that we've used a frame.
+        self.window.has_frame.store(false, Ordering::Relaxed);
 
-        let should_draw = self.window.should_draw.clone();
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        if let Some(surface) = self.window.wayland_surface() {
+            let has_frame = self.window.has_frame.clone();
+            // Request a new frame.
+            surface.frame().quick_assign(move |_, _, _| {
+                has_frame.store(true, Ordering::Relaxed);
+            });
 
-        // Mark that window was drawn.
-        should_draw.store(false, Ordering::Relaxed);
+            return;
+        }
 
-        // Request a new frame.
-        surface.frame().quick_assign(move |_, _, _| {
-            should_draw.store(true, Ordering::Relaxed);
-        });
+        // Get the display vblank interval in micro seconds.
+        let monitor_vblank_interval = Duration::from_micros(
```

Comment:
Why are you using microseconds here? I find that extremely strange considering basically everyone uses either ms or nanos usually.

Just for the sub-millis accuracy without having to use nanos which would be overly precise?

Reply:
That's a good question. Simply ns is not enough, so I've picked next... Could use nanos if you feel like it.

Reply:
Micros should be fine I guess. The explanation makes enough sense where people shouldn't be completely confused reading this.